### PR TITLE
net-im/skype-dbus-mock: mock system dbus so skype works without it

### PR DESCRIPTION
**Pull request by logistiker**
Bug: https://bugs.gentoo.org/669538
Signed-off-by: Matthew Schultz <mattsch@gmail.com>
Package-Manager: Portage-2.3.62, Repoman-2.3.11

### CODE REVIEW

**Comment by juippis** [comment]
I just noticed the bug in Bugzilla. Could you make suggested modifications to skypeforlinux in this PR (you'll have to rebase against fresh master, too). Then we get new skypeforlinux maintainer to review it too.

I just want to be sure nothing breaks and everyone agrees with this change.

**Comment by logistiker** [comment]
It's a simple one line addition to skypeforlinux in RDEPEND which would look like this:
```
        || (
                sys-auth/elogind
                sys-apps/systemd
                net-im/skype-dbus-mock
        )
```

**Comment by logistiker** [comment]
Thanks for taking notice of this too!  I'm tired of updating my local portage copy of skypeforlinux every time it upgrades.

File: net-im/skype-dbus-mock/skype-dbus-mock-0_pre20181011.ebuild [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python{2_7,3_4,3_5,3_6,3_7} )
```
Comment (juippis) [inline net-im/skype-dbus-mock/skype-dbus-mock-0_pre20181011.ebuild]: Take python3.4 off, it's not in portage anymore.

File: net-im/skype-dbus-mock/skype-dbus-mock-0_pre20181011.ebuild [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+RDEPEND=">=net-im/skypeforlinux-8.30
+	dev-python/dbus-python[${PYTHON_USEDEP}]
+	!sys-apps/systemd
```
Comment (juippis) [inline net-im/skype-dbus-mock/skype-dbus-mock-0_pre20181011.ebuild]: Make this and elogind a hard blocker. Portage can "solve" these soft blocks on its own which might surprise users, hard blocker will stop them if they have systemd or elogind installed.

Rest looks good. You'll need to fix the CI issues too.

**Comment by juippis** [comment]
@jvesely please take a look if you accept changes to skypeforlinux or not.

File: net-im/skype-dbus-mock/skype-dbus-mock-0_pre20181011.ebuild (hunk 2 above)
Comment (mgorny) [reply net-im/skype-dbus-mock/skype-dbus-mock-0_pre20181011.ebuild]: I think soft blocker is fine here. We really only use hard blockers when something needs to be unmerged prior to building other package. They're PITA to users.

Comment (mgorny) [reply net-im/skype-dbus-mock/skype-dbus-mock-0_pre20181011.ebuild]: (and if someone uses systemd or logind, the appropriate USE flags will bring them in anyway)

Comment (logistiker) [reply net-im/skype-dbus-mock/skype-dbus-mock-0_pre20181011.ebuild]: Ok, so there's nothing to be done here then?  Which USE flags were you referring to?

Comment (mgorny) [reply net-im/skype-dbus-mock/skype-dbus-mock-0_pre20181011.ebuild]: Many packages have USE=systemd and/or USE=elogind, generally to cover optional support. I have no strong opinion whether skype should have those flags as well -- I suppose `|| ()` dep works fine here.

**Comment by logistiker** [comment]
I'm not sure how to solve the NonsolvableDepsInDev QA problem.

**Comment by juippis** [comment]
Since skypeforlinux is masked here, 
https://github.com/gentoo/gentoo/blob/master/profiles/arch/amd64/x32/package.mask

you'll need to add skype-dbus-mock to that list as well. Also as I asked, please rebase on top of fresh gentoo, because you are editing an ebuild for skypeforlinux that doesnt exist in Gentoo tree anymore.

**Comment by logistiker** [comment]
Sorry, I thought I did rebase my branch.  I'll base it off the 138 ebuild.

**Comment by juippis** [comment]
Ok, LGTM. I won't wait for @jvesely's comment now because I won't have time to check these on weekend, but if I see bug reports about this in the morning I'll revert the skypeforlinux edit. 

Thanks!

**Comment by juippis** [comment]
FYI, you can have automated sign-off when using repoman by setting DCO_SIGNED_OFF_BY in your make.conf. Your 2nd commits sign-off was malformed.

**Comment by logistiker** [comment]
Ah thanks for the tip about DCO_SIGNED_OFF_BY.  I didn't know there was something automatic for that.

**Pull request closed without merging**